p2p.go

```diff
@@ -263,7 +264,8 @@ type Host struct {
 	kadKey         cid.Cid
 	newPubSub      func(ctx context.Context, h core.Host, opts ...pubsub.Option) (*pubsub.PubSub, error)
 	pubsub         *pubsub.PubSub
-	pubs           map[string]*pubsub.Topic
+	pubs           sync.Map
```

Comment:
do we need a `sync.Map`?

Reply:
if the answer is yes, `subs` at L270 also need `sync.Map`?

Reply:
we can just use a `mutex` to protect `pubs` and `subs` from concurrent accessing